### PR TITLE
src/emmc: check the return value of g_fopen

### PR DESCRIPTION
**Pull request by jluebbe**
This could be NULL on error, so handle that case.

### CODE REVIEW

File: src/emmc.c [hunk 1]
```diff
@@ -156,6 +156,12 @@ static gboolean r_emmc_force_part_write(const gchar *device, gchar value, GError
 	}
 
 	f = g_fopen(sysfs_path, "w");
+	if (!f) {
```
Comment (ejoerns) [inline src/emmc.c:159]: Shouldn't we evaluate `errno` here to provide more targeted error messages?

Comment (jluebbe) [reply src/emmc.c:159]: Yes, and we could merge it with the g_file_test above, which becomes redundant with this check.

File: src/emmc.c [hunk 2]
```diff
@@ -148,15 +148,14 @@ static gboolean r_emmc_force_part_write(const gchar *device, gchar value, GError
...
 
 	sysfs_path = g_strdup_printf("/sys/block/%s/force_ro", device_basename);
-	if (!g_file_test(sysfs_path, G_FILE_TEST_EXISTS)) {
+	f = g_fopen(sysfs_path, "w");
+	if (!f) {
 		g_set_error(error, R_UPDATE_ERROR, R_UPDATE_ERROR_FAILED,
-				"Block device does not have force_ro attribute: %s does not exist",
-				sysfs_path);
+				"Could not open device attribute %s: %s",
+				sysfs_path, g_strerror(errno));
```
Comment (ejoerns) [inline src/emmc.c:155]: Please save `errno` with `int err = errno;` as suggested by glib documentation and as we do for other similar calls, too.

Comment (jluebbe) [reply src/emmc.c]: I've adjusted the other calls in `emmc.c` as well.

**Review by ejoerns — APPROVED** [review]
Looks fine for me now.

**Review by Bastian-Krause — APPROVED** [review]

**Pull request merged**